### PR TITLE
[stable/consul] Promote initContainers to podspec

### PR DESCRIPTION
**Pull request by dhilipkumars**
Move initcontainers to podspec

### CODE REVIEW

**Comment by dhilipkumars** [comment]
cc: @lachie83 PTAL

File: stable/consul/templates/consul-test.yaml [hunk 1]
```diff
@@ -4,21 +4,20 @@ metadata:
...
-          {"name": "tools", "mountPath": "/tools"}
-        ]
-      }
-    ]'
 spec:
+  initContainers:
+    - name: test-framework
+      image: dduportal/bats:0.4.0
+      command:
+        - "ba$h"
```
Comment (prydonius) [inline stable/consul/templates/consul-test.yaml]: typo

File: stable/consul/templates/consul-test.yaml [hunk 2]
```diff
@@ -4,21 +4,20 @@ metadata:
...
+        - "-c"
+        - |
+          set -ex
+          # copy bats to tools dir
+          cp -R /usr/local/libexec/ /tools/bats/
+      volumeMounts:
+      - mountPath: /tools
```
Comment (prydonius) [inline stable/consul/templates/consul-test.yaml]: inconsistent indentation for lists here, this line should probably indented another 2 spaces

**Review by prydonius — CHANGES_REQUESTED** [review]

**Comment by dhilipkumars** [comment]
@prydonius Thanks for the review PTAL now.

**Review by prydonius — APPROVED** [review]

**Comment by prydonius** [comment]
cc @lachie83 the tests appear to be failing here

**Comment by prydonius** [comment]
@viglesiasce since we delete the namespace, we're not getting logs if tests fail. Helm test prints out a command to get the logs when tests fail, maybe we can parse that and execute it?

**Comment by unguiculus** [comment]
I guess CI doesn't wait long enough. Chart installs fine on my GKE cluster and tests run successfully.

**Pull request merged**